analysis/quantifier-comparisons/01-import-quant-data.Rmd

```diff
@@ -0,0 +1,235 @@
+samples = c('SCPCR000001', 'SCPCR000002')
+
+# generate include statements for sample directories
+includes = stringr::str_glue("--include \"*/{samples}*\"")
+
+# Alevin quantification files
+sync_call <- paste('aws s3 sync', quant_s3_alevin, alevin_data_dir,
+                   includes)
+system(sync_call, ignore.stdout = TRUE)
+
+# Kallisto quantification files
+# exclude bus files, which are large
+sync_call <- paste('aws s3 sync', quant_s3_kallisto, kallisto_data_dir,
+                   includes, '--exclude "*/bus/*"')
+system(sync_call, ignore.stdout = TRUE)
+
+# Cellranger quantification files
+# exclude intermediate and bam files
+sync_call <- paste('aws s3 sync', quant_s3_cellranger, cellranger_data_dir,
+                   includes, 
+                   '--exclude "*/SC_RNA_COUNTER_CS/*"',
+                   '--exclude "*.bam"', '--exclude "*.bam.bai"')
+system(sync_call, ignore.stdout = TRUE)
+```
+
+
+### Generate sample info data frame
+
+This generates the sample list from the directory names of the downloads. 
+
+```{r}
+
+quant_info <- c(alevin = alevin_data_dir, 
+                kallisto = kallisto_data_dir,
+                cellranger = cellranger_data_dir) %>%
+  purrr::map(list.dirs, recursive = FALSE, full.names = FALSE) %>%
+  # convert to data frame
+  tibble::enframe(name = "tool", value = "quant_dir") %>%
+  tidyr::unnest(quant_dir) %>%
+  # parse the `quant_dir` into sample and index info
+  tidyr::separate(quant_dir, 
+                  sep = "[-]",
+                  into = c("sample", "index_type"),
+                  remove = FALSE) %>%
+  tidyr::separate(index_type, 
+                  into = c("index_content", "kmer", "decoy"), 
+                  extra = "drop",
+                  fill = "right") %>%
+  # no decoy samples get NA here, convert to text.
+  dplyr::mutate(decoy = tidyr::replace_na(decoy, "no"))
+
+# save info table
+readr::write_tsv(quant_info, quant_info_file )
+# print the info table
+quant_info
+
+```
+
+## Process quantification files
+
+### Alevin
+Use tximport to make SCEs
+
+```{r}
+alevin_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "alevin") %>%
+  dplyr::pull(quant_dir)
+
+alevin_sces <- alevin_quant_ids  %>%
+  purrr::map(
+    ~ tximport(file.path(alevin_data_dir, .x, "alevin", "quants_mat.gz"),
+               type = "alevin")) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x$counts))) 
+
+# add names
+names(alevin_sces) <- alevin_quant_ids
+```
+
+### Kallisto
+
+Get file basenames (no extension) for import
+```{r}
+kallisto_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "kallisto") %>%
+  dplyr::pull(quant_dir)
+
+base_file <- file.path(kallisto_data_dir, kallisto_quant_ids, "counts", "gene_count")
+```
+
+#### Function for reading Kallisto files
+
+Takes a base file name and reads the the `.mtx` file and associated `.barcodes.txt` and `.genes.txt` files.  
+Optionally filters cells/barcodes based on the knee plot using `DropletUtils::barcodeRanks()`.
```

Comment:
I would add a sentence here that is explicit about what this is attempting to accomplish and/or what the knee plot _plots_. You do make reference to filtering for empty cells in the next sentence.

Reply:
I have added some discussion of knee plots in e28aa5875d721b36977b8f6a04c6fc9895b4c71c (also adding a bit of info to the function body)
Let me know if that clarifies sufficiently.